src/main/java/fi/otavanopisto/kuntaapi/server/integrations/casem/CasemFileEntityUpdater.java

```diff
@@ -0,0 +1,233 @@
+
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private CaseMTranslator casemTranslator;
+  
+  @Inject
+  private IdController idController;
+
+  @Inject
+  private IdentifierController identifierController;
+
+  @Inject
+  private CasemFileCache fileCache;
+  
+  @Inject
+  private ModificationHashCache modificationHashCache;
+
+  @Inject
+  private CaseMFileController casemFileController;
+
+  @Inject
+  private Event<IndexRequest> indexRequest;
+
+  @Inject
+  private Event<IndexRemoveRequest> indexRemoveRequest;
+  
+  @Resource
+  private TimerService timerService;
+
+  private boolean stopped;
+  private IdUpdateRequestQueue<FileIdUpdateRequest> queue;
+
+  @PostConstruct
+  public void init() {
+    queue = new IdUpdateRequestQueue<>(CaseMConsts.IDENTIFIER_NAME);
+  }
+
+  @Override
+  public String getName() {
+    return "casem-files";
+  }
+
+  @Override
+  public void startTimer() {
+    startTimer(TIMER_INTERVAL);
+  }
+
+  private void startTimer(int duration) {
+    stopped = false;
+    TimerConfig timerConfig = new TimerConfig();
+    timerConfig.setPersistent(false);
+    timerService.createSingleActionTimer(duration, timerConfig);
+  }
+
+  @Override
+  public void stopTimer() {
+    stopped = true;
+  }
+  
+  @Asynchronous
+  public void onPageIdUpdateRequest(@Observes FileIdUpdateRequest event) {
+    if (!stopped) {
+      FileId fileId = event.getId();
+      
+      if (!StringUtils.equals(fileId.getSource(), CaseMConsts.IDENTIFIER_NAME)) {
+        return;
+      }
+      
+      queue.add(event);
+    }
+  }
+  
+  @Asynchronous
+  public void onPageIdRemoveRequest(@Observes FileIdRemoveRequest event) {
+    if (!stopped) {
+      FileId fileId = event.getId();
+      
+      if (!StringUtils.equals(fileId.getSource(), CaseMConsts.IDENTIFIER_NAME)) {
+        return;
+      }
+      
+      deleteFile(event, fileId);
+    }
+  }
+  
+  @Timeout
+  public void timeout(Timer timer) {
+    if (!stopped) {
+      FileIdUpdateRequest updateRequest = queue.next();
+      if (updateRequest != null) {
+        updateCasemFile(updateRequest);
+      }
+
+      startTimer(SystemUtils.inTestMode() ? 1000 : TIMER_INTERVAL);
+    }
+  }
+  
+  private void updateCasemFile(FileIdUpdateRequest updateRequest) {
+    OrganizationId organizationId = updateRequest.getOrganizationId();
+    FileId casemFileId = updateRequest.getId();
+    PageId casemPageId = updateRequest.getPageId();
+    
+    DownloadMeta downloadMeta = casemFileController.getDownloadMeta(casemFileId);
+    if (downloadMeta != null) {
+      updateCasemFile(casemFileId, casemPageId, downloadMeta, updateRequest.getOrderIndex());
+    } else {
+      logger.log(Level.SEVERE, () -> String.format("Organization %s file %s meta could not be downloaded", organizationId.getId(), casemFileId.toString()));
+    }
+  }
+  
+  private void updateCasemFile(FileId casemFileId, PageId casemPageId, DownloadMeta downloadMeta, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateCasemFile" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)